app/components/viral/modal_component.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Viral
+  # A component for displaying a modal.
+  class ModalComponent < Component
+    attr_reader :title
```

Comment:
**chore (blocking):** Currently, by looking at the preview, there is only one size modal and it takes up most of the screen.  The one feature I would like you to add is sizes.  Can you implement it the same way that the `dropdown_component.rb` handles `TRIGGER_MAPPINGS`.  You can use the styles from [Flowbite modal sizes](https://flowbite.com/docs/components/modal/#sizes).  Make sure you set a default size.

Reply:
Please see 8c9b83bf72a3527e4bbb5c2b451f3501f0485d39.